### PR TITLE
Dismiss options for snv cancer variants

### LINKED ISSUE #1614: Dismissed variants are not grey
From https://github.com/Clinical-Genomics/scout/issues/1460

### PR DESCRIPTION
**Pull request by mayabrandi**
This PR adds 3 cancer-specific dismiss options to the select for dismissing variants on the cancer variant page.

Fixes the code so that dismissed snv cancer variants appear grey on the cancer variants page (fix 
#1614)
**How to test**:
1. Install a local implementation and load the demo cancer case with the following command: `scout --demo load case scout/demo/cancer.load_config.yaml`.

1. Go to cancer case and dismiss a variant with a specific cancer option (the 3 options at the end of the list in the dismiss select). 
1. Go back to the cancer variants page and make sure that the variant's line is grey

1. Check that dismissing a variant from the demo case (non-cancer) works and that the 3 cancer specific options are not present in the dismiss select of this case.

**Expected outcome**:
The functionality should be working as described.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN

### CODE REVIEW

**Comment by dnil** [comment]
Yeah, agreed: either a new set of dismiss options, or probably better a member variable (list) telling which variant types this dismiss option can apply to.

File: scout/constants/__init__.py [hunk 1]
```diff
@@ -15,8 +15,9 @@
 from .gene_tags import GENE_CUSTOM_INHERITANCE_MODELS
 from .variant_tags import (CONSEQUENCE, CONSERVATION, DISMISS_VARIANT_OPTIONS,
                            FEATURE_TYPES, SPIDEX_LEVELS, SPIDEX_HUMAN, SV_TYPES,
-                           GENETIC_MODELS, VARIANT_CALL,
-                           CANCER_TIER_OPTIONS, MANUAL_RANK_OPTIONS, MOSAICISM_OPTIONS)
+                           GENETIC_MODELS, VARIANT_CALL, CANCER_TIER_OPTIONS,
+                           CANCER_SPECIFIC_VARIANT_DISMISS_OPTIONS, MANUAL_RANK_OPTIONS,
```
Comment (dnil) [inline scout/constants/__init__.py]: 😄 👍

File: scout/constants/variant_tags.py [hunk 2]
```diff
@@ -229,6 +229,24 @@
...
 
+CANCER_SPECIFIC_VARIANT_DISMISS_OPTIONS = {
+    44: {
+        'label': 'Possible Germline',
+        'description': 'Variant is possibly a germline event.',
+        'evidence': []
+        },
+    45: {
+        'label': 'Low count normal',
+        'description': 'If normal sample exists, variant has too few reads in normal sample "AD".',
```
Comment (dnil) [inline scout/constants/variant_tags.py]: I can't follow this one: how about just
```variant has too few reads in normal sample "AD".```.

File: scout/server/blueprints/variant/controllers.py [hunk 3]
```diff
@@ -181,6 +182,10 @@ def variant(
             overlapping_vars.append(var)
     variant_obj["end_chrom"] = variant_obj.get("end_chrom", variant_obj["chromosome"])
 
+    dismiss_options = DISMISS_VARIANT_OPTIONS
+    if case_obj.get('track') == 'cancer':
+        dismiss_options = {**DISMISS_VARIANT_OPTIONS, **CANCER_SPECIFIC_VARIANT_DISMISS_OPTIONS}
```
Comment (dnil) [inline scout/server/blueprints/variant/controllers.py]: :+1:

**Comment by northwestwitch** [comment]
Ready for review!

File: scout/server/blueprints/variants/templates/variants/cancer-variants.html [hunk 4]
```diff
@@ -63,7 +63,11 @@
       </thead>
       <tbody>
         {% for variant in variants %}
-          <tr>
+          {% if variant.dismiss_variant %}
+            <tr class="dismiss">
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/cancer-variants.html]: :+1:

**Comment by dnil** [comment]
Nice! This works locally:
Screenshot 2020-01-28 at 09 22 58
Screenshot 2020-01-28 at 09 28 56
Screenshot 2020-01-28 at 09 28 23

File: scout/constants/variant_tags.py (hunk 2 above)
Comment (northwestwitch) [reply scout/constants/variant_tags.py]: Ok! I don't know where that description comes from .. Probably you, a few months ago! :)

Comment (dnil) [reply scout/constants/variant_tags.py]: I'm sure! 😆

**Comment by northwestwitch** [comment]
Don't merge yet. I'm also checking the general report and it looks like something has to be fixed there as well

**Comment by dnil** [comment]
Ok, it did look good so I assumed the controller change was sufficient.

**Comment by northwestwitch** [comment]
> Ok, it did look good so I assumed the controller change was sufficient.

Eh, the 3 option have to be added to the dismiss options there as well, otherwise it will crash

**Comment by northwestwitch** [comment]
Now it should work!

**Comment by northwestwitch** [comment]
Oh no, coveralls, really?

**Comment by northwestwitch** [comment]
Coverage is up and I'm going to merge in 3 .. 2 .. 1 ..

**Pull request merged**